render/dom-for.js

```diff
@@ -0,0 +1,24 @@
+"use strict"
+
+var delayedRemoval = module.exports.delayedRemoval = new WeakMap
```

Comment:
`const`, surely?

Reply:
Indeed. This was written before I committed to ditch OOTB IE11 compat

Reply:
Same thing in the tests — top level references are all `var`s. I figure if the PR is internally consistent, we can procedurally transform the rest of the codebase at a later stage.

Reply:
Oh, no, that was a plain copy/paste, then aping what was present... I'll update the test as well, var/let/const mixed is dissonant.

Reply:
The bundler doesn't like destructuring and const the generator is apparently fine. Bailing out of most of ES6 syntax for now.